decentdao/decent-interface
# Configurable fractal module attachment

## Changes

- Add Checkbox into Child Safe creation flow to allow user control whether `FractalModule` should be deployed & enabled on Child Safe or not
- Adjust Child Safe creation logic accordingly to that setting
- Do not display "Initiate clawback" option from `ManageDAOMenu` when `FractalModule` is not attached to the Child Safe

Closes #2385 

## Testing

1. Create Parent Safe (Multisig or any other type - doesn't really matter)
2. Create Child Safe with "Enable Clawback" checkbox being checked (it is checked by default)
3. Navigate to Child Safe settings page
4. Make sure `FractalModule` appears in the list of Safe Modules
5. Freeze Child Safe
6. Make sure "Initiate clawback" option appears in the `ManageDAOMenu` once Child Safe is frozen
7. Create Child Safe with "Enable Clawback" checkbox being unchecked (make sure to hit this buddy)
8. Navigate to 2nd Child Safe settings page
9. Make sure `FractalModule` does not appear in the list of Safe Modules
10. Freeze 2nd Child Safe
11. Make sure "Initiate clawback" option does not appear in the `ManageDAOMenu` once Child Safe is frozen
12. Celebrate

## Updated UI

<img width="941" alt="Screenshot 2024-10-01 at 17 37 21" src="https://github.com/user-attachments/assets/30e01d98-237b-4a6c-82c5-fc3b372fe463">
